common/persistence/persistenceInterface.go

```diff
@@ -46,8 +46,30 @@ type (
 		GetShard(ctx context.Context, request *InternalGetShardRequest) (*InternalGetShardResponse, error)
 		UpdateShard(ctx context.Context, request *InternalUpdateShardRequest) error
 	}
+
 	// TaskStore is a lower level of TaskManager
-	TaskStore = TaskManager
+	TaskStore interface {
+		Closeable
+		GetName() string
+		LeaseTaskList(ctx context.Context, request *LeaseTaskListRequest) (*LeaseTaskListResponse, error)
```

Comment:
`LeaseTaskListRequest` takes a TaskListKind which is an int. That in represents a value of a thrift enum. Which is a serialization detail. Therefore `LeaseTaskListRequest` should be separated.

Reply:
The same is true of `LeaseTaskListResponse`

Reply:
There are other cases in which this is true as well please take a another pass over those request and response structs. I think many of them have taskListKind. We will want to represent task list kind as an application level enum eventually.

Reply:
The taskListKind is an enum in dataInterface. So, I didn't understand how it is related to serialization here.